File: count_sort/CountSort1.java
class CountSort1{
    static int[] countSort1(int[] A){
        int max=A[0];
        for(int i=0;i<A.length;i++){
            if(A[i]>max){
                max=A[i];
            }
        }
        int[] C=new int[max+1];
        for (int i=0;i<A.length;i++){
            C[A[i]]++;
        }

        int Ai=0;
        for(int i=0;i<C.length;i++){
            if(C[i]!=0){
                A[Ai]=i;
                C[i]--;
                i--;
                Ai+=1;
            }
        }
        return A;
    }
    public static void main(String[] args){
        int[] A={
            1, 8, 6, 3, 10, 9
        };
        A=countSort1(A);
        for(int i=0;i<A.length;i++){
            System.out.print(A[i]+"\t");
        }
    }
}

File: count_sort/count_sort1.py
def count_sort1(A):
    max_=A[0]
    for i in range(len(A)):
        if(max_<A[i]):
            max_=A[i]
            
    C = [0]*(max_+1)
    for i in range(len(A)):
        C[A[i]]+=1
    
    Ai = 0
    for i in range(len(C)):
        if(C[i]!=0):
            A[Ai]=i
            Ai+=1
            
    return A

print(count_sort1([1, 2, 10, 8, 7]))